File: server_integration_tests/java/java_server/src/main/java/dev/graphlink/javaserver/impl/ReservedFieldsServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.enums.Keyword;
import dev.graphlink.javaserver.generated.inputs.ReservedInput;
import dev.graphlink.javaserver.generated.services.ReservedFieldsService;
import dev.graphlink.javaserver.generated.types.Class;
import dev.graphlink.javaserver.generated.types.ReservedFields;
import dev.graphlink.javaserver.generated.types.Secret_;
import org.springframework.stereotype.Service;

@Service
public class ReservedFieldsServiceImpl implements ReservedFieldsService {

    private ReservedFields.Builder base() {
        return ReservedFields.builder()
            .id("r1")
            .class_("cls")
            .return_(42)
            .new_(true)
            .default_("def")
            .is("yes")
            .in("inside")
            .with("w")
            .int_(7)
            .synchronized_(false)
            .native_("n")
            .kind(Keyword.CLASS)
            .nested(Class.builder().id("c1").value("v").build())
            .secret(Secret_.builder().id("s1").token("tok").build());
    }

    @Override
    public ReservedFields reserved() {
        return base().build();
    }

    @Override
    public ReservedFields switch_(String class_, Integer return_) {
        return base().class_(class_).return_(return_ != null ? return_ : 0).build();
    }

    @Override
    public ReservedFields echoReserved(ReservedInput input) {
        return base()
            .class_(input.getClass_())
            .return_(input.getReturn_() != null ? input.getReturn_() : -1)
            .default_(input.getDefault_() != null ? input.getDefault_() : "none")
            .build();
    }
}

File: server_integration_tests/kotlin/kotlin_server_blocking/src/main/kotlin/dev/graphlink/kotlinserverblocking/impl/ReservedFieldsServiceImpl.kt
package dev.graphlink.kotlinserverblocking.impl

import dev.graphlink.kotlinserverblocking.generated.enums.Keyword
import dev.graphlink.kotlinserverblocking.generated.inputs.ReservedInput
import dev.graphlink.kotlinserverblocking.generated.services.ReservedFieldsService
import dev.graphlink.kotlinserverblocking.generated.types.Class
import dev.graphlink.kotlinserverblocking.generated.types.ReservedFields
import dev.graphlink.kotlinserverblocking.generated.types.Secret_
import org.springframework.stereotype.Service

@Service
class ReservedFieldsServiceImpl : ReservedFieldsService {

    private fun base(): ReservedFields = ReservedFields(
        id = "r1",
        class_ = "cls",
        return_ = 42,
        new = true,
        default = "def",
        is_ = "yes",
        in_ = "inside",
        with = "w",
        int = 7,
        synchronized = false,
        native = "n",
        kind = Keyword.CLASS,
        nested = Class(id = "c1", value = "v"),
        secret = Secret_(id = "s1", token = "tok"),
    )

    override fun reserved(): ReservedFields = base()

    override fun switch(class_: String, return_: Int?): ReservedFields =
        base().copy(class_ = class_, return_ = return_ ?: 0)

    override fun echoReserved(input: ReservedInput): ReservedFields = base().copy(
        class_ = input.class_,
        return_ = input.return_ ?: -1,
        default = input.default ?: "none",
    )
}
